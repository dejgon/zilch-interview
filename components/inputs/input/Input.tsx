import React from 'react';
import { Text, TextInput, TextInputProps, View } from 'react-native';

import useStyles from './styles';

interface Props extends TextInputProps {
  disabled?: boolean;
  errorMessage?: string;
  isError?: boolean;
  label?: string;
}

export function Input(props: Props) {
  const { label, isError, errorMessage, disabled, ...rest } = props;

  const {
    disabledInput,
    error,
    inputBase,
    inputContainer,
    inputLabel,
    labelError,
  } = useStyles;

  return (
    <View testID="input-view-container" style={[inputContainer, rest.style]}>
      {label && (
        <Text testID="input-label" style={inputLabel}>
          {label}
        </Text>
      )}
      <TextInput
        {...rest}
        testID="input-text-area"
        style={{
          ...inputBase,
          ...(disabled && disabledInput),
          ...(isError && error),
        }}
        editable={!disabled}
      />
      {isError && (
        <Text style={{ ...inputLabel, ...(isError && labelError) }}>
          {errorMessage}
        </Text>
      )}
    </View>
  );
}
