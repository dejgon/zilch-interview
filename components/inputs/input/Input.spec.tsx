import React from 'react';

import { render } from '@testing-library/react-native';

import { Input } from './Input';

describe('Input.tsx', () => {
  it('should render basic snapshot', () => {
    const component = render(
      <Input
        value="Basic component"
        label="Email address"
        keyboardType="email-address"
        placeholder="Enter your email address"
        autoCapitalize="none"
        disabled={false}
        autoCorrect={false}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should render error', () => {
    const { getByText } = render(
      <Input
        value="Error component"
        label="Email address"
        keyboardType="email-address"
        placeholder="Enter your email address"
        autoCapitalize="none"
        disabled={false}
        autoCorrect={false}
        isError
        errorMessage="Error message"
      />
    );

    expect(getByText('Error message')).toBeDefined();
  });

  it('should render disabled', () => {
    const { getByTestId } = render(
      <Input
        value="Error component"
        label="Email address"
        keyboardType="email-address"
        placeholder="Enter your email address"
        autoCapitalize="none"
        disabled
        autoCorrect={false}
      />
    );

    expect(getByTestId('input-text-area').props.editable).toBeFalsy();
  });
});
