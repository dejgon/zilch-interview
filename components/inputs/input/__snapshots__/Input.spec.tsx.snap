// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.tsx should render basic snapshot 1`] = `
<View
  style={
    {
      "paddingBottom": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#313131",
        "fontFamily": "Oswald-Regular",
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 21,
        "paddingBottom": 4,
        "paddingLeft": 4,
      }
    }
  >
    Email address
  </Text>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    editable={true}
    keyboardType="email-address"
    placeholder="Enter your email address"
    style={
      {
        "borderColor": "#313131",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "#313131",
        "fontFamily": "Oswald-Regular",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "400",
        "height": 48,
        "lineHeight": 24,
        "paddingLeft": 8,
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`Input.tsx should render disabled 1`] = `
<View
  style={
    {
      "paddingBottom": 16,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#313131",
        "fontFamily": "Oswald-Regular",
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "lineHeight": 21,
        "paddingBottom": 4,
        "paddingLeft": 4,
      }
    }
  >
    Email address
  </Text>
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    editable={false}
    keyboardType="email-address"
    placeholder="Enter your email address"
    style={
      {
        "borderColor": "#808080",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "#808080",
        "fontFamily": "Oswald-Regular",
        "fontSize": 16,
        "fontStyle": "normal",
        "fontWeight": "400",
        "height": 48,
        "lineHeight": 24,
        "paddingLeft": 8,
        "width": "100%",
      }
    }
  />
</View>
`;
